gnucash/register/register-gnome/gnucash-sheet-private.c

```diff
@@ -479,6 +479,20 @@ draw_cell (GnucashSheet *sheet,
 
     text = gnc_table_get_entry (table, virt_loc);
 
+    // use glyths for the associate entries if possible
+    if (sheet->use_glyphs)
+    {
+        const gchar *name = gnc_table_get_cell_name (table, virt_loc);
+
+        if (g_strcmp0 (name, "associate") == 0 && g_strcmp0 (text, " ") != 0)
+        {
+            if (g_strcmp0 (text, "f") == 0) // only have two options
+                text = GLYPH_PAPERCLIP;
+            else
+                text = GLYPH_LINK;
+        }
```

Comment:
Can `text` ever be NULL in this function or is it always one of " ", "f" or "l" ?

If it's never null it's likely to be more efficient to test `*text != ' '` and `*text == 'f'` rather than going through g_strcmp0. If it can be NULL you may have to guard the whole test with a `if (text)`

Reply:
A bigger issue here is though that I don't think this is the spot to do this. You have added code in the generic sheet that now needs to distinguish based on the presence of a very specific column. The sheet is supposed to be column type agnostic.
Instead you should define a new cell type for the association that does this glyph replacement for you and returns either a single letter or a glyph as text when requested for it.